cassandraprotocol/constants.go

```diff
@@ -260,3 +208,15 @@ const (
 	RowsFlagNoMetadata       = RowsFlag(0x00000004)
 	RowsFlagMetadataChanged  = RowsFlag(0x00000008)
 )
+
+type VariablesFlag = int32
+
+const (
+	VariablesFlagGlobalTablesSpec = VariablesFlag(0x00000001)
```

Comment:
I tried to figure out what this is but I couldn't, can you give me a brief explanation?

Reply:
It's a reference to the sole flag defined for `PreparedResult` bound variables metadata, see [here](https://github.com/datastax/go-cassandra-native-protocol/blob/main/specs/native_protocol_v4.spec#L653).

For a bit of context: the Java equivalent of this project defines one single object called `RowsMetadata` and applies it to both `PreparedResult` and `RowsResult` messages in 3 places:

1. To model the `PreparedResult` `<metadata>` structure
2. To model the `PreparedResult` `<result_metadata>` structure
3. To model the `RowsResult` `<metadata>` structure

Indeed, 2 and 3 are identical (barring the fact that not all flags are meaningful depending on the message), but in fact, 1 is not 100% identical to 2 and 3. 

After struggling with this in Go for a while, I decided that it was easier to create a separate struct for 1 above, which is why in this PR you see a `RowsMetadata` (for 2 and 3 above) and also a new `VariablesMetadata` struct (for 1 above). For the same reason, I preferred to create a separate `VariablesFlag` that represents the flags of a `VariablesMetadata` struct.

Reply:
Thanks for the explanation! I've never looked too much into the prepared metadata so this is very useful 👍